ui/src/app/pages/analysis/gke-app-launcher.spec.tsx

```diff
@@ -0,0 +1,149 @@
+import '@testing-library/jest-dom';
+
+import { MemoryRouter } from 'react-router';
+import { Route } from 'react-router-dom';
+import { mockNavigate } from 'setupTests';
+
+import {
+  AppStatus,
+  AppType,
+  ListAppsResponse,
+  WorkspaceAccessLevel,
+} from 'generated/fetch';
+
+import { screen } from '@testing-library/dom';
+import { render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { GKEAppLauncher } from 'app/pages/analysis/gke-app-launcher';
+import { analysisTabPath, appDisplayPath } from 'app/routing/utils';
+import { currentWorkspaceStore } from 'app/utils/navigation';
+import { userAppsStore } from 'app/utils/stores';
+
+import { workspaceStubs } from 'testing/stubs/workspaces';
+
+const RSTUDIO_FAKE_URL = 'https://fakeRStudioUrl/';
+const createApp = (appType: AppType = AppType.RSTUDIO) => [
+  {
+    googleProject: 'googleProject',
+    appName: 'rstudioApp',
+    appType: appType,
+    status: AppStatus.RUNNING,
+    proxyUrls: { app: RSTUDIO_FAKE_URL },
+  },
+];
+
+const setUpWorkspace = () => {
+  currentWorkspaceStore.next({
+    ...workspaceStubs[0],
+    accessLevel: WorkspaceAccessLevel.WRITER,
+  });
+};
+const currentWorkspace = workspaceStubs[0];
+
+const createRoute = (appType: string) => (
+  <MemoryRouter
+    initialEntries={[
+      appDisplayPath(currentWorkspace.namespace, currentWorkspace.id, appType),
+    ]}
+  >
+    <Route path='/workspaces/:ns/:wsid/analysis/userApp/:appType'>
+      <GKEAppLauncher hideSpinner={() => {}} showSpinner={() => {}} />
+    </Route>
+  </MemoryRouter>
+);
+
+const pathToCurrentWSAnalysisTab = analysisTabPath(
+  currentWorkspace.namespace,
+  currentWorkspace.id
+);
+
+const setup = (
+  queryParam: string = 'RStudio',
+  app: ListAppsResponse = createApp()
+) => {
+  userAppsStore.set({
+    updating: false,
+    userApps: app,
+  });
+  setUpWorkspace();
+  return {
+    container: render(createRoute(queryParam)).container,
+    user: userEvent.setup(),
+  };
+};
+
+const setupEmptyUserStore = () => {
+  userAppsStore.set({
+    updating: false,
+    userApps: [],
+  });
+
+  return {
+    container: render(createRoute('RStudio')).container,
+    user: userEvent.setup(),
+  };
+};
+
+it('Should display iframe with rStudio App Url', async () => {
+  setup();
+  // Confirm IFrame exist
+  expect(screen.getByTitle('RStudio embed')).toBeInTheDocument();
+  // Confirm IFrame source is same as proxyUrl for Rstudio L24
+  expect(screen.getByTitle('RStudio embed').getAttribute('src')).toBe(
+    RSTUDIO_FAKE_URL
+  );
+});
+
+it('Should display error message if App type is not found', async () => {
+  setup('fakeApp');
+  expect(
+    screen.getByText(
+      /an error was encountered with your fakeapp environment\. to resolve, please see the applications side panel\./i
```

Comment:
This is a potential security vulnerability - we should only detect that the user-provided text is invalid, not attempt to process it for display.